### PR TITLE
Document client-level configuration

### PR DESCRIPTION
**Pull request by florimondmanca**
Somewhat related to #343.

### CODE REVIEW

File: docs/advanced.md [hunk 1]
```diff
@@ -15,6 +15,42 @@ A Client instance is equivalent to a Session instance in `requests`.
...
+'app-1.0'
+```
+
+!!! note
+    When you provide a parameter at both the client and request levels, one of two things can happen:
+    
+    - For headers, query parameters and cookies, the values are merged into one.
+    - For all other parameters, the request-level value is used.
+
+Besides, `Client` constructor accepts some parameters that aren't available at the request level.
```
Comment (yeraydiazdiaz) [inline docs/advanced.md]: I'm not a native english speaker but "besides" feels argumentative, I feel "additionally" may read better?

Maybe something like "Additionally, the `Client` constructor..."

File: docs/advanced.md [hunk 2]
```diff
@@ -15,6 +15,42 @@ A Client instance is equivalent to a Session instance in `requests`.
...
+...
+>>> r.json()['headers']['User-Agent']
+'app-1.0'
+```
+
+!!! note
+    When you provide a parameter at both the client and request levels, one of two things can happen:
+    
+    - For headers, query parameters and cookies, the values are merged into one.
+    - For all other parameters, the request-level value is used.
```
Comment (yeraydiazdiaz) [inline docs/advanced.md]: Excellent 👏

File: docs/advanced.md [hunk 3]
```diff
@@ -36,6 +36,42 @@ Alternatively, you can explicitly close the connection pool without block-usage
...
+For example, to apply a set of custom headers on every request:
+
+```python
+>>> url = 'http://httpbin.org/headers'
+>>> headers = {'user-agent': 'my-app/0.0.1'}
+>>> with httpx.Client(headers=headers) as client:
+...     r = client.get(url)
+...
+>>> r.json()['headers']['User-Agent']
+'app-1.0'
```
Comment (yeraydiazdiaz) [inline docs/advanced.md]: Just tried this snippet locally and the result is 'my-app/0.0.1'?

Comment (florimondmanca) [reply docs/advanced.md]: Oh, right!

Comment (florimondmanca) [inline docs/advanced.md]:
```suggestion
'my-app/0.0.1'
```

**Pull request merged**